docs/source/API/core/policies/RangePolicy.rst

```diff
@@ -23,41 +23,49 @@ Synopsis
 
 .. code-block:: cpp
 
+    struct Kokkos::ChunkSize {
+        ChunkSize(int value_);
+    };
+
     template<class ... Args>
-    class Kokkos::RangePolicy {
-        typedef RangePolicy execution_policy;
-        typedef typename traits::index_type member_type ;
-        typedef typename traits::index_type index_type;
+    struct Kokkos::RangePolicy {
+        using execution_policy = RangePolicy;
+        using member_type = PolicyTraits<Args...>::index_type;
 
-        //Inherited from PolicyTraits<Args...>
+        // Inherited from PolicyTraits<Args...>
         using execution_space   = PolicyTraits<Args...>::execution_space;
         using schedule_type     = PolicyTraits<Args...>::schedule_type;
         using work_tag          = PolicyTraits<Args...>::work_tag;
         using index_type        = PolicyTraits<Args...>::index_type;
         using iteration_pattern = PolicyTraits<Args...>::iteration_pattern;
         using launch_bounds     = PolicyTraits<Args...>::launch_bounds;
 
-        //Constructors
+        // Constructors
         RangePolicy(const RangePolicy&) = default;
         RangePolicy(RangePolicy&&) = default;
 
-        inline RangePolicy();
+        RangePolicy();
+
+        RangePolicy( member_type work_begin
+                   , member_type work_end );
 
-        template<class ... Args>
-        inline RangePolicy( const execution_space & work_space
-                          , const member_type work_begin
-                          , const member_type work_end
-                          , Args ... args);
+        RangePolicy( member_type work_begin
+                   , member_type work_end
+                   , ChunkSize chunk_size );
 
-        template<class ... Args>
-        inline RangePolicy( const member_type work_begin
-                          , const member_type work_end
-                          , Args ... args);
+        RangePolicy( const execution_space & work_space
+                   , member_type work_begin
+                   , member_type work_end );
+
+        RangePolicy( const execution_space & work_space
+                   , member_type work_begin
+                   , member_type work_end
+                   , ChunkSize chunk_size );
 
         // retrieve chunk_size
-        inline member_type chunk_size() const;
+        member_type chunk_size() const;
         // set chunk_size to a discrete value
-        inline RangePolicy set_chunk_size(int chunk_size_);
+        RangePolicy& set_chunk_size(int chunk_size_);
 
         // return ExecSpace instance provided to the constructor
         KOKKOS_INLINE_FUNCTION const execution_space & space() const;
```

Comment:
drop the inline "specifiers" below (KOKKOS_FUNCTION instead of KOKKOS_INLINE_FUNCTION)